src/wallet/coin_selection.rs

```diff
@@ -117,11 +106,14 @@ use crate::types::{FeeRate, UTXO};
 /// overridden
 pub type DefaultCoinSelectionAlgorithm = LargestFirstCoinSelection;
 
+/// The weight of a transaction input with an empty witness and scriptSig
+pub const TXIN_BASE_WEIGHT: usize = 164;
```

Comment:
It looks like @danielabrozzoni 's PR has done something similar here (and chosen the exact same varaible name 😎).

I think decomposing the value as there is a good idea.

Reply:
I've added the description [here] (without the comment about electrs because I'm not sure what it is getting at).

[here]: https://github.com/bitcoindevkit/bdk/pull/148/files#diff-fd2a47357bd088bed85419710e86d949d1a172691db87c978be1336a725a42afR249

Reply:
In that other PR the "BASE_WEIGHT" had been originally set to `prev_txid (32 bytes) + prev_vout (4 bytes) + sequence (4 bytes)` without counting the "+1" for the script len, because @danielabrozzoni assumed that it was included in the `max_satisfaction_weight` by rust-miniscript. This was making the end to end tests with the full node and electrs fail because the fees were just below min-relay-fee, because the weight was being underestimated by exactly 1 vbyte.

I'm still kinda thinking this could be a bug in rust-miniscript, because potentially longer scripts can require more than 1 byte for the size, since it should be a VarInt. But in the meantime I guess we just have to workaround it and add our own "+1".

Reply:
Maybe instead of the comment we should open an issue, just to make sure we don't forget about it

Reply:
I'll remove the `FIXME:` in my PR and open an issue then :)